### PR TITLE
chore: update version using script

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

versions can be really different from one page to another

**What is the chosen solution to this problem?**

create a script used in the CI to sync versions of dependencies used

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

- [x] Add script version.js to update versions in each package.json in all the stack
- [x] Add all tools (babel, webpack, ...) used accross the stack
- [x] Add option to update deps of a project using UI
- [x] Add option to update the stack in a project

### CODE REVIEW

File: packages/cmf/__tests__/route.test.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('CMF route', () => {
 });
 
 describe('loadComponent behavior', () => {
-	it('should inject dispatch into component properties from context.store', () => {
+	xit('should inject dispatch into component properties from context.store', () => {
```
Comment (talend-glorieux) [inline packages/cmf/__tests__/route.test.js]: Please fix this test.

File: packages/cmf/package.json [hunk 2]
```diff
@@ -79,21 +79,21 @@
...
+    "redux": "3.6.0",
+    "redux-batched-actions": "0.2.0",
+    "redux-thunk": "2.2.0"
   },
   "version": "0.74.2",
   "dependencies": {
     "hoist-non-react-statics": "^1.2.0"
   }
-}
+}
```
Comment (talend-glorieux) [inline packages/cmf/package.json]: Looks like all the package.json files have a missing new line at end of file.

File: version.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	.option('-d, --debug', 'display more info')
+	.option('-q, --quiet', 'display nothing')
+	.option('-p, --path [value]', 'path of the package.json to update')
+	.parse(process.argv);
+
+const REACT_VERSION = '15.5.4';
+const JEST_VERSION = '20.0.3';
+
+const VERSIONS = {
+	// deps
```
Comment (talend-glorieux) [inline version.js]: Unnecessary comment.

File: version.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	}
+	return modified;
+}
+
+function checkAll(source, dep) {
+	const version = VERSIONS[dep];
+	const devDeps = source.devDependencies;
+	const deps = source.dependencies;
+	const peer = source.peerDependencies;
+	const mDevDeps = check(devDeps, dep, version);
```
Comment (talend-glorieux) [inline version.js]: Why the `m` prefix?

File: version.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	if (!program.quiet) {
+		console.log(`save ${ppath}`);
+	}
+	fs.open(ppath, 'w', function(err, fd) {
+		if (err) {
+				throw 'error opening file: ' + err;
+		}
+
+		fs.write(fd, data, 0, data.length, null, function(err) {
+			if (err) throw 'error writing file: ' + err;
```
Comment (talend-glorieux) [inline version.js]: missing brackets around if statement.

File: version.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	Object.keys(VERSIONS).forEach((dep) => {
+		checkAll(packageJSON, dep);
+	});
+	if (packageJSON.modified) {
+		delete packageJSON.modified;
+		save(ppath, JSON.stringify(packageJSON, null, 2));
+		console.log(path.join(path.dirname(ppath), 'yarn.lock'));
+		// TODO: remove yarn.lock using fs.unlink();
```
Comment (talend-glorieux) [inline version.js]: Is that TODO still valid?

File: .travis/after_success_git.sh [hunk 7]
```diff
@@ -23,6 +23,25 @@ if [ -n "$GH_TOKEN" ]; then
...
+		git add packages/logging/package.json
+		git add packages/theme/package.json
+		git add packages/cmf/yarn.lock
+		git add packages/components/yarn.lock
+		git add packages/containers/yarn.lock
+		git add packages/forms/yarn.lock
+		git add packages/generator/yarn.lock
+		git add packages/icons/yarn.lock
+		git add packages/logging/yarn.lock
+		git add packages/theme/yarn.lock
```
Comment (talend-glorieux) [inline .travis/after_success_git.sh]: I'd rather see a loop used here to avoid copy errors.

File: version.js (hunk 4 above)
Comment (jmfrancois) [inline version.js]: isModified is better

File: version.js (hunk 3 above)
Comment (jmfrancois) [inline version.js]: not agree: this is for dev who want to understand the reading of the file.

File: version.js (hunk 6 above)
Comment (jmfrancois) [inline version.js]: removed

File: version.js (hunk 3 above)
Comment (jsomsanith) [inline version.js]: Agree

**Comment by jmfrancois** [comment]
I have opened #482 to keep only the script in a first step

**Pull request closed without merging**